quickwit-actors/src/actor_handle.rs

```diff
@@ -188,7 +188,13 @@ impl<A: Actor> ActorHandle<A> {
 
     /// Waits until the actor exits by itself. This is the equivalent of `Thread::join`.
     pub async fn join(self) -> (ActorExitStatus, A::ObservableState) {
-        let exit_status = self.join_handle.join().await;
+        let exit_status = self.join_handle.await.unwrap_or_else(|join_err| {
+            if join_err.is_panic() {
+                ActorExitStatus::Panicked
+            } else {
+                ActorExitStatus::Killed
```

Comment:
just for your info, in tokio, the join error is either panic or a cancellation. Killed seems a bit harsh :)

Reply:
Killed is correct.

Names are for the client of our API. 
We want a name that matches quickwit's naming here, not tokio's naming.

This task being cancelled can only happen if an actor got killed.